File: source/back-end/src/main/java/be/pxl/backend/service/DestinationService.java
package be.pxl.backend.service;

import java.util.List;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import be.pxl.backend.entity.Destination;
import be.pxl.backend.repository.DestinationRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@Transactional
public class DestinationService implements IDestinationService {

	@Autowired
	private DestinationRepository repo;
	
	public Destination find(int id) {
		return repo.findOne(id);
	}
	
	public List<Destination> all() {
		return repo.findAll();
	}
	
	public void persist(Destination destination) {
		repo.save(destination);
	}
	
	public void delete(int id) {
		repo.delete(id);
	}
	
	public void update(Destination destination) {
		Destination d = repo.findOne(destination.getDestination_id());
		d.copy(destination);
		repo.save(destination);
	}
	
}

File: Source/frontend/frontend.Service/DestinationService.cs
using frontend.Domain;
using frontend.Repository;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace frontend.Service
{
    public class DestinationService : IDestinationService
    {
        private DestinationRepository destinationRepository;

        public DestinationService()
        {
            destinationRepository = new DestinationRepository();
        }

        public void Add(Destination destination)
        {
            destinationRepository.AddDestination(destination);
        }

        public List<Destination> All()
        {
            return destinationRepository.GetAllDestinations().Result.ToList();
        }

        public void Delete(int id)
        {
            destinationRepository.DeleteDestination(id);
        }

        public Destination Find(int id)
        {
            return destinationRepository.GetDestinationById(id).Result;
        }

        public void Update(Destination destination)
        {
            destinationRepository.UpdateDestination(destination);
        }
    }
}
